.github/workflows/test.yml

```diff
@@ -0,0 +1,61 @@
+name: Test
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+on:
+  # Trigger flow when pushing in main or pull requests, and when creating
+  # a pull request.
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'packages/react-native-sdk/docusaurus/**'
+      - 'packages/react-sdk/docusaurus/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths-ignore:
+      - 'packages/react-native-sdk/docusaurus/**'
+      - 'packages/react-sdk/docusaurus/**'
+
+jobs:
+  test-and-build:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+
+      - name: ESLint Cache
+        uses: actions/cache@v3
+        with:
+          path: './.eslintcache'
+          key: ${{ runner.os }}-eslintcache-${{ github.ref_name }}-${{ hashFiles('.eslintcache') }}
+
+      - name: Install Dependencies
+        run: yarn install --immutable
+
+      - name: Lint Packages
+        run: yarn lint:ci:all
+
+      - name: Build packages
+        run: NODE_ENV=production yarn build:all
+
+      - name: Test packages
+        run: yarn test:ci:all
```

Comment:
Can we run the tests before we build everything? I don't think it makes sense to wait until all apps finish building and then stop because we have a failing test somewhere...

Reply:
I've changed the order